src/Sulu/Bundle/WebsiteBundle/Controller/CacheController.php

```diff
@@ -24,16 +27,21 @@ class CacheController extends Controller
     /**
      * Clear the whole http_cache for website.
      *
+     * @param Request $request
+     *
      * @return JsonResponse
      */
-    public function clearAction()
+    public function clearAction(Request $request)
     {
         if (!$this->checkLivePermissionForAllWebspaces()) {
             return new JsonResponse(null, 403);
         }
 
         $this->get('sulu_website.http_cache.clearer')->clear();
 
+        $cacheEvent = new CacheEvent($request);
```

Comment:
why do you pass the request here? do you need it for your usecase? and i think we should raise the event in the service. /cc @danrot @chirimoya

Reply:
I don't necessarily need the request here (yet). When needed I could get request-related info from inside the eventsubscriber, so I can remove it.

The CacheClearer::clear() is also called from other controllers (e.g. `AnalyticsController`), that's why I wanted to be more explicit and only raise the event in the CacheController, when the user actually clicks on Settings/Clear cache. 

(As a side-note: the `CacheClearer service is getting a little bloated with dependencies.)

However, if you still think it better to dispatch the event from within the Service, I'll move it there, and then I will remove the `Events` and `CacheEvent` classes as obsolete.

Reply:
I would also argue that putting the event dispatching into the service is a good idea. If you want to execute something when the cache is cleared, you probably also want to do it in the `AnalyticsController`. Or at least you would expect it to be ran always.

Reply:
True. Changed.